sheetstorm/tests/test_unit_generate_test_data.py

```diff
@@ -0,0 +1,259 @@
+        self.assertEqual(GenerateTestDataCommand._set_user_factory_parameters(user_type).get("user_type"), user_type)
+
+    @parameterized.expand(
+        [
+            (CustomUser.UserType.ADMIN.name, 2),
+            (CustomUser.UserType.EMPLOYEE.name, 5),
+            (CustomUser.UserType.MANAGER.name, 3),
+        ]
+    )
+    def test_result_of_create_user_function_should_be_specified_number_of_users_in_database(
+        self, user_type, number_of_users_to_create
+    ):
+        GenerateTestDataCommand().create_user(user_type, number_of_users_to_create)
+
+        self.assertEqual(CustomUser.objects.filter(user_type=user_type).count(), number_of_users_to_create)
+
+    @parameterized.expand(
+        [
+            (CustomUser.UserType.ADMIN.name, 2, 2),
+            (CustomUser.UserType.EMPLOYEE.name, 4, 3),
+            (CustomUser.UserType.MANAGER.name, 2, 1),
+        ]
+    )
+    def test_create_user_function_should_not_create_users_if_number_in_database_is_greater_than_or_equal_to_specified(
+        self, user_type, mock_number_of_users_in_database, number_of_users_to_create
+    ):
+        GenerateTestDataCommand().create_user(user_type, mock_number_of_users_in_database)
+
+        GenerateTestDataCommand().create_user(user_type, number_of_users_to_create)
+
+        self.assertEqual(CustomUser.objects.filter(user_type=user_type).count(), mock_number_of_users_in_database)
+
+
+class CreateProjectMethodsTests(TestCase):
+    def setUp(self):
+        self.mock_number_of_suspended_projects_to_create = 4
+        self.mock_number_of_active_projects_to_create = 5
+        self.mock_number_of_completed_projects_to_create = 7
+
+        self.options = {
+            ProjectType.SUSPENDED.name: self.mock_number_of_suspended_projects_to_create,
+            ProjectType.ACTIVE.name: self.mock_number_of_active_projects_to_create,
+            ProjectType.COMPLETED.name: self.mock_number_of_completed_projects_to_create,
+        }
+
+        self.suspended_projects = self._generate_test_projects(
+            number_of_projects_to_fill_database=2, project_type=ProjectType.SUSPENDED.name
+        )
+        self.number_of_suspended_projects_to_create = self.options[ProjectType.SUSPENDED.name] - len(
+            self.suspended_projects
+        )
+
+        self.active_projects = self._generate_test_projects(
+            number_of_projects_to_fill_database=2, project_type=ProjectType.ACTIVE.name
+        )
+        self.number_of_active_projects_to_create = self.options[ProjectType.ACTIVE.name] - len(self.active_projects)
+
+        self.completed_projects = self._generate_test_projects(
+            number_of_projects_to_fill_database=2, project_type=ProjectType.COMPLETED.name
+        )
+        self.number_of_completed_projects_to_create = self.options[ProjectType.COMPLETED.name] - len(
+            self.completed_projects
+        )
```

Comment:
I still think there should be tests for cases of creating projects when the database is empty and not all tests here require presence of projects in the database. Because of that, I would lean towards creating objects only in the few tests that require it. In that case, I would move it to the method to be able to call it conveniently in said tests.

Reply:
Done